node/resolvers/profile/index.ts

```diff
@@ -44,6 +52,37 @@ export const mutations = {
 export const queries = {
   profile: (_: any, { customFields }: any, context: Context) =>
     getProfile(context, customFields),
+
+  checkProfileAllowed: async (_: any, __: any, context: Context) => {
+    const {
+      clients: { catalog, session },
+      vtex: { segment },
+      cookies,
+    } = context
+    const salesChannel = segment ? segment.channel : null
+
+    const { sessionData } = await session.getSession(
+      cookies.get('vtex_session')!,
+      ['*']
+    )
+
+    const email: string | undefined = path(
+      ['namespaces', 'profile', 'email', 'value'],
+      sessionData
+    )
+
+    if (!email) {
+      return { allowed: null }
```

Comment:
Would it make sense to return `allowed: false` in this case, since you could not validate that the user is allowed? Then, by default, a user would not be allowed to access whatever you're validating in the front-end.